admintools/middleware.py

```diff
@@ -18,7 +18,7 @@ def process_request(self, request):
                     request.user = emulated_user
 
     def process_response(self, request, response):
-        if not (hasattr(request, 'original_user') or can_emulate_another_user(request.user)):
+        if not hasattr(request, 'original_user') or can_emulate_another_user(request.user):
```

Comment:
I think this is wrong.

if you want to do this, you have to express

`not A and not B`

or

`not(A or B)`